### PR TITLE
LKE-7645 fix(init viz): await async call when initiating a viz

### PR DESCRIPTION
**Pull request by aminerhanemi**

### CODE REVIEW

**Review by gmasclet — COMMENTED** [review]
Looks good overall :slightly_smiling_face: 

If we really want to be sure we don't forget to handle a promise, I recommend adding the [no-floating-promises](https://typescript-eslint.io/rules/no-floating-promises/) rule.

I'm not convinced turning the `no-empty-function` and `require-await` rules into error ones is the right move:
* Empty functions and async functions with no await are not really harmful as far as I know. There are even sometimes compelling reasons to have async functions with no await.
* We're better off having consistent linting rules all across our code-base, so I suggest not changing them without a good reason.

**Comment by aminerhanemi** [comment]
I am curious to know in which case an async function make sense without an await

File: .eslintrc.js [hunk 1]
```diff
@@ -23,10 +23,11 @@ module.exports = {
     '@typescript-eslint/ban-ts-ignore': ['off'], // Allows @ts-ignore
     '@typescript-eslint/interface-name-prefix': ['off'], // Allows interfaces prefixed with I
     '@typescript-eslint/no-non-null-assertion': ['off'], // Allows non-null assertion
-    '@typescript-eslint/no-empty-function': ['off'], // Allows empty functions
-    '@typescript-eslint/require-await': ['off'], // Allows async without await
+    '@typescript-eslint/no-empty-function': ['error'], // Allows empty functions
```
[the remark this answers is not in the document]
Comment (aminerhanemi) [reply .eslintrc.js:26]: I will revert it as it is not relevant with the context of the PR

**Comment by gmasclet** [comment]
@aminerhanemi here is an example: https://eslint.org/docs/latest/rules/require-await#when-not-to-use-it

Otherwise, I note that an async function is also convenient when you want to return a mix of promises and non-promise values:

```typescript
async function foo(): Promise<Item[]> {
  if (someGuardCondition()) {
    // This value is transparently converted to a promise because foo is async
    return [];
  }
  return bar();
}

function bar(): Promise<Item[]> {
  // some computation that asynchronously returns an array of items
}
```

**Comment by aminerhanemi** [comment]
@gmasclet I think you copied the wrong example.
I saw that it can make sense if you want to use promise error catch mechanism. I don't think we have this use case in the actual code base, so I think it will be fine to keep the rule enabled.

**Pull request merged**